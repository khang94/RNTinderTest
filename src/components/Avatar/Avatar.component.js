import React from 'react';
import PropTypes from 'prop-types';
import {Image, View} from 'react-native';

import styles from './Avatar.styles';

const Avatar = ({urlAvatar}) => (
  <View style={styles.containerAvatar}>
    <View style={styles.avatarWrapper}>
      <Image style={styles.avatar} source={{uri: urlAvatar}} />
    </View>
  </View>
);

Avatar.propTypes = {
  urlAvatar: PropTypes.string.isRequired,
};

export default Avatar;
