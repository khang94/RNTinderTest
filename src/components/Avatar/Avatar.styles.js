import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  containerAvatar: {
    alignItems: 'center',
  },
  avatarWrapper: {
    borderRadius: 150,
    borderWidth: 0.5,
    borderColor: '#757575',
  },
  avatar: {
    borderRadius: 150,
    borderWidth: 5,
    borderColor: '#ffffff',
    width: 150,
    height: 150,
    resizeMode: 'contain',
  },
});
