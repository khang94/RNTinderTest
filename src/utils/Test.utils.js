import React from 'react';
import renderer from 'react-test-renderer';

const singleSnapTest = (tree, description) => {
  it(description, () => {
    expect(tree).toMatchSnapshot();
  });
};

export const testSnapshots = (Component, configs) => {
  describe('snapshots', () => {
    configs.forEach(config => {
      const {props, description} = config;

      const tree = renderer.create(<Component {...props} />);
      singleSnapTest(tree, description);
    });
  });
};
