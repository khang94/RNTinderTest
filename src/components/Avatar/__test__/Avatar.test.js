import Avatar from '../Avatar.component';
import {testSnapshots} from '../../../utils/Test.utils';

describe('Avatar snapshot', () => {
  testSnapshots(Avatar, [
    {
      props: {
        urlAvatar: 'http://api.randomuser.me/landscape/male/2323.jpg',
      },
      description: 'default renderer',
    },
  ]);
});
