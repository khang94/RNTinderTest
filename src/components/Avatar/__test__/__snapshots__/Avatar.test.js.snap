// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshot snapshots default renderer 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <View
    style={
      Object {
        "borderColor": "#757575",
        "borderRadius": 150,
        "borderWidth": 0.5,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "http://api.randomuser.me/landscape/male/2323.jpg",
        }
      }
      style={
        Object {
          "borderColor": "#ffffff",
          "borderRadius": 150,
          "borderWidth": 5,
          "height": 150,
          "resizeMode": "contain",
          "width": 150,
        }
      }
    />
  </View>
</View>
`;
